Automattic/wp-calypso
# General fixes of the Blaze Dashboard integration with Jetpack

This PR fixes some issues regarding the integration of the Blaze Advertising dashboard inside Jetpack (using the Blaze Dashboard app).

## Proposed Changes

* Fix the double scroll shown on the Create campaign widget. This also fixes the footer and part of the menu showing at the bottom of the modal
![Screenshot 2023-06-27 at 15 46 35](https://github.com/Automattic/wp-calypso/assets/1258162/ad85cd4b-6378-4560-8ad1-e8767edec95f)

* Fix the wrong spacing on the notices. For example, the rejection notice shown in the details page for a rejected campaign.
<img width="1394" alt="jetpack-campaign-details" src="https://github.com/Automattic/wp-calypso/assets/1258162/58c68f15-7172-41eb-90c1-74b3a267228a">

* Fix the support links not redirect the user to the support page. Now, for Jetpack sites using the Blaze Dashboard app, the support links open in a new tab.
**The links that changed were:**
Learn more in the header
Promote link in the No campaigns message
Learn more in credits page
View documentation in the campaign details support links section.

* Now that the Advertising page changed the Powered label in the details page, I am showing it also here (it was hidden in a previous commit).
![Screenshot 2023-06-27 at 15 50 14](https://github.com/Automattic/wp-calypso/assets/1258162/fd6365d1-420a-4dea-acc0-a267c39a8685)


## Testing Instructions

You will need a local Jetpack installation to test this. 

* Checkout this branch, then open a terminal and navigate to `wp-calypso/apps/blaze-dashboard`
* Execute `BLAZE_DASHBOARD_PACKAGE_PATH=/path/to/jetpack/projects/packages/blaze yarn dev`
You will need to change the path to your local Jetpack installation.
* In another terminal tab, go to the Jetpack repo directory and execute: `pnpm jetpack build packages/blaze`
* Now open your Jurassic tunnel and enter the wp-admin page for your local site
* Click on Tools->Advertising 
**Note:** If you don't see that option, you will need to execute a Code snippet that enable blaze for your site.
-- Install the Plugin `Code Snippets`
-- Create a new snippet with the following content: `add_filter( 'jetpack_blaze_dashboard_enable', '__return_true' );`

* Now that you are in the Advertising page, verify that the **Learn more** link on the Header opens the correct tab to the support page.
* Click on the Credits tab, and verify the same for the **Learn more** link in there.
* Click on the Promote button of any post
* Verify that you don't see two scrolls, and by scrolling down the page you don't see the menu or footer. 
(I had to shrink my window height to half the size to be able to reproduce this on prod, if your window size is too bug).
* Close the widget and go to the details of a rejected campaign
* Verify that the spacing of the notice of top is correct
* Verify that in the bottom of the page, the `Blaze - Powered by Jetpack` label is shown.



## Pre-merge Checklist
